zerver/tests/test_decorators.py

```diff
@@ -2343,3 +2363,373 @@ def view_A(
             [b'{"result":"success","msg":"","ignored_parameters_unsupported":["author"]}\n'],
         )
         self.assert_json_success(result, ignored_parameters=["author"])
+
+
+ParamTypes = Literal["none", "json_only", "both"]
+
+
+class TestEndpoint(ZulipTestCase):
```

Comment:
This test file is kinda huge. It might be a helpful prep PR to split `test_decorators.py` up. I think we could do:
* `test_validators.py` for `ValidatorTestCase` to parallel `zerver/lib/validator.py`
* And then tests might go into a new `test_api_parameter_parsing.py` (and maybe your new `api.py` would be better named `zerver/lib/api_parameter_parsing.py` or something?

(There's other things we might be able to split out, like a `test_auth_decorators.py`, but I'm not certain of what a good division might be and that may be more clear after this project)

Reply:
Yeah, the name `api.py` is tentative.

Reply:
I think some test cases are included in `test_decorators` not because they test decorators, but because they are used in one of the authentication decorators. It is pretty unclear.